src/settings/config.rs

```diff
@@ -0,0 +1,89 @@
+// Config file handling
+
+use std::env;
+
+use serde::Deserialize;
+
+use crate::{dimensions::Dimensions, frame::Frame};
+
+use std::path::{Path, PathBuf};
+
+const CONFIG_FILE: &str = "config.toml";
+
+#[cfg(unix)]
+fn neovide_config_dir() -> PathBuf {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("neovide").unwrap();
+    xdg_dirs.get_config_home()
+}
+
+#[cfg(windows)]
+fn neovide_config_dir() -> PathBuf {
+    let mut path = dirs::config_dir().unwrap();
+    path.push("neovide");
+    path
+}
+
+pub fn config_path() -> PathBuf {
+    let mut config_path = neovide_config_dir();
+    config_path.push(CONFIG_FILE);
+    config_path
+}
+
+#[derive(Debug, Deserialize, Default)]
+pub struct Config {
+    pub multigrid: Option<bool>,
+    pub maximized: Option<bool>,
+    pub vsync: Option<bool>,
+    pub srgb: Option<bool>,
+    pub no_idle: Option<bool>,
+    pub neovim_bin: Option<PathBuf>,
+    pub frame: Option<Frame>,
+    pub geometry: Option<Dimensions>,
+}
+
+impl Config {
+    pub fn write_to_env(&self) {
```

Comment:
by the way, is there any alternatives to setting variables from config?

Reply:
What do you mean by alternatives?

Reply:
I mean, instead of writing to env, maybe we there is some other way? But it it works, and there is no better way, it's fine.